README.md

```diff
@@ -1,48 +1,47 @@
 # intercom-php
 
-[![Code
-Climate](https://codeclimate.com/repos/537da4a7e30ba062b101be9c/badges/2aa25d4736f09f40282e/gpa.svg)](https://codeclimate.com/repos/537da4a7e30ba062b101be9c/feed) [![Circle CI](https://circleci.com/gh/intercom/intercom-php.png?style=badge)](https://circleci.com/gh/intercom/intercom-php)
+[![Code Climate](https://codeclimate.com/repos/537da4a7e30ba062b101be9c/badges/2aa25d4736f09f40282e/gpa.svg)](https://codeclimate.com/repos/537da4a7e30ba062b101be9c/feed) [![Circle CI](https://circleci.com/gh/intercom/intercom-php.png?style=badge)](https://circleci.com/gh/intercom/intercom-php)
 
 Official PHP bindings to the Intercom API
 
 ## Installation
 
-Requires PHP 5.6.
+Requires PHP 5.6 or above.
 
-The recommended way to install intercom-php is through [Composer](https://getcomposer.org):
-
-First, install Composer:
-
-```sh
-curl -sS https://getcomposer.org/installer | php
-```
+This library uses [HTTPPlug](https://github.com/php-http/httplug) as HTTP client. HTTPPlug is an abstraction that allows this library to support many different HTTP Clients. Therefore, you need to provide it with an adapter for the HTTP library you prefer. You can find the available adapters [here](http://docs.php-http.org/en/latest/clients.html). This documentation assumes you use the Guzzle6 Client, but you can replace it with any adapter that you prefer.
 
-Next, install the latest intercom-php:
+The recommended way to install intercom-php is through [Composer](https://getcomposer.org):
 
 ```sh
-php composer.phar require intercom/intercom-php
-```
-
-Finally, you need to require the library in your PHP application:
-
-```php
-require "vendor/autoload.php";
+composer require intercom/intercom-php php-http/guzzle6-adapter
 ```
 
 ## Clients
 
-For OAuth or Access Tokens use:
+Initialize your client using your access token:
 
 ```php
 use Intercom\IntercomClient;
 
-$client = new IntercomClient('<insert_token_here>', null);
+$client = new IntercomClient('<insert_token_here>');
 ```
 
 > If you already have an access token you can find it [here](https://app.intercom.com/a/apps/_/developer-hub). If you want to create or learn more about access tokens then you can find more info [here](https://developers.intercom.com/building-apps/docs/authorization#section-access-tokens).
 >
 > If you are building a third party application you can get your OAuth token by [setting-up-oauth](https://developers.intercom.com/building-apps/docs/authorization#section-oauth) for Intercom.
 
+For most of the use cases, the code snippet above is just enough. However, for most specific scenarios, you can include custom request headers, or even pass in your own HTTP client, request factory or URI factory:
+
+```php
+use Intercom\IntercomClient;
+
+$client = new IntercomClient('<insert_token_here>', null, ['Custom-Header' => 'value']);
+
+$client->setClient($myCustomHttpClient); // $myCustomHttpClient implements Psr\Http\Client\ClientInterface
```

Comment:
`$client->setClient` is a bit confusing. Could this be called `setHttpClient` to more clearly differentiate it from the Intercom client?

Reply:
Great point! I agree this is confusing, I wanted to keep the method name unchanged to reduce friction on the upgrade but actually, as the parameter type has changed, it won't hurt to rename this.

Reply:
Oh and don't forget to change the release notes too!

Reply:
Done 😄 Thanks a lot for the review! 🙌